plugins/action/action.h

```diff
@@ -24,29 +38,164 @@ class Action
         UNKNOWN
     };
 
+    /**
+     * Returns a human-readable English string for an Action::Result.
+     */
     static const char *result_str(Result);
 
-    typedef std::function<void(Result)> result_callback_t;
-
+    /**
+     * Arms the drone (synchronous).
+     *
+     * Note that arming a drone normally means that the motors will spin at idle.
+     * Therefore, before arming take all safety precautions and stand clear of the drone.
+     *
+     * @return result of request
+     */
     Result arm() const;
+
+    /**
+     * Disarms the drone (synchronous).
```

Comment:
Is there some sort of timeout on synchronous requests? When is UNKNOWN sent? What should the user do in that case?

Reply:
In my opinion, I would leave it as is.

Reply:
What about item "2"?

Reply:
Sorry, forgot to asnwer 2:

I think it should be mostly caught by the errors. It could be that sometimes we would get a denied without further information (unless you listen for mavlink status text which we currently don't). I would say we improve the docs as we go and find edge cases.